geetools/ee_image_collection.py

```diff
@@ -1044,12 +1047,32 @@ def reduceInterval(
         # create a reducer from user parameters
         red = getattr(ee.Reducer, reducer)() if isinstance(reducer, str) else reducer
 
+        # to avoid different behaviour between single and multi-band image we hard coded the
+        # band names to be suffixed with the reducer output names.
+        def outputNames(b):
+            return red.getOutputs().map(lambda o: ee.String(b).cat("_").cat(o))
+
+        bandNames = self._obj.first().bandNames().map(outputNames).flatten()
+
+        # original behaviour is keeping in the output the original band names which is not compatible with
+        # muli-output reducer for obvious reasons. We need to change this back to the original band names
+        # suffixed with the reducer output names to avoid non-regression issue we added the
+        # "keep_original_names" parameter that will be removed in downstream version
+        if keep_original_names is True and red.getOutputs().length().getInfo() == 1:
```

Comment:
I imagine a user trying to keep original names in a multi-output reducer.. shouldn't her/him be warned?

Reply:
they were not able to launch the method anyway so I think they should not be warned about a temporary parameter.